rsyncd/rsyncd.go

```diff
@@ -25,22 +25,34 @@ type sendTransfer struct {
 	lastMatch int64
 }
 
-func NewServer(modules ...config.Module) *Server {
-	return &Server{modules}
+type Module struct {
+	Name string   `toml:"name"`
+	Path string   `toml:"path"`
+	ACL  []string `toml:"acl"`
+}
+
+func NewServer(modules ...Module) (*Server, error) {
```

Comment:
This can now fail due to added validation

Reply:
I commented about this in earlier comment (you wanted to remove NewModule() for extensibility, I think that makes validation more important)

Reply:
BTW I'm not super sold on `...Module` instead of `[]Module`, as that rules out future option-arguments.

The original rationale was that some rsyncd library callers may have hardcoded modules so it just looks nicer when you don't need a slice.

(Maybe this is non-issue if you don't have issues breaking backwards compat before v1 release.)

Reply:
Fair point. Let’s use `[]Module` here, also because a zero-module server is not generally useful, so having `rsyncd.NewServer()` fail at compile-time is actually desired.